python/hail/api2/matrixtable.py

```diff
@@ -2031,6 +2031,42 @@ def filter_cols_list(self, keys, keep=True):
             jkeys.append(ck_type._convert_to_j(k))
         return MatrixTable(self._jvds.filterSamplesList(jkeys, keep))
 
+    @handle_py4j
+    @typecheck_method(mapping=listof(strlike))
+    def reorder_columns(self, mapping):
+        """Reorder columns.
+
+        Examples
+        --------
+
+        Randomly shuffle order of samples:
+
+        >>> import random
+        >>> new_sample_order = vds.sample_ids[:]
+        >>> random.shuffle(new_sample_order)
+        >>> vds_reordered = vds.reorder_samples(new_sample_order)
+
+        Notes
+        -----
+
+        This method requires unique sample ids. `mapping` must contain the
+        same ids as :meth:`~hail.VariantDataset.sample_ids`. The order of the
+        ids in `mapping` determines the sample id order in the output dataset.
```

Comment:
Can you make it clear that the full columns, with their field and entry values, are reordered (this is not just renaming).